agnpy/absorption/absorption.py

```diff
@@ -350,6 +357,71 @@ def evaluate_tau_dt(
         prefactor = (L_disk * xi_dt) / (8 * np.pi ** 2 * epsilon_dt * m_e * c ** 3)
         return (prefactor * integral).to_value("")
 
+    @staticmethod
+    def evaluate_tau_dt_mu_s(
+        nu,
+        z,
+        mu_s,
+        L_disk,
+        xi_dt,
+        epsilon_dt,
+        R_dt,
+        r,
+        u_size=100,
+        phi_re=phi_to_integrate,
+    ):
+        r"""Evaluates the gamma-gamma absorption produced by a ring dust torus
+        for the case of photon moving at an angle to the jet 
+
+        Parameters
+        ----------
+        nu : :class:`~astropy.units.Quantity`
+            array of frequencies, in Hz, to compute the sed 
+            **note** these are observed frequencies (observer frame)
+        z : float
+            redshift of the source
+        mu_s : float
+            cosine of the angle between the blob motion and the jet axis
+        L_disk : :class:`~astropy.units.Quantity`
+            Luminosity of the disk whose radiation is being reprocessed by the BLR
+        xi_dt : float
+            fraction of the disk radiation reprocessed by the disk
+        epsilon_dt : string
+            peak (dimensionless) energy of the black body radiated by the torus 
+        R_dt : :class:`~astropy.units.Quantity`
+            radius of the ting-like torus
+        r : :class:`~astropy.units.Quantity`
+            distance between the dust torus and the blob
+        u_size : int
+            size of the array of distances from the photon origin to integrate over
+        phi_re : :class:`~numpy.ndarray`
+            arrays of azimuth angles of the dust torus to integrate over
+
+        Returns
+        -------
+        :class:`~astropy.units.Quantity`
+            array of the SED values corresponding to each frequency
+        """
+        # conversions
+        epsilon_1 = nu_to_epsilon_prime(nu, z)
+        # multidimensional integration
+        # here uu is the distance that the photon traversed
+        uu = np.logspace(-5, 5, u_size) * r
```

Comment:
Why do you start integrating from `1e-5 * r`, just for me to know because I always go from `r` to `1e5 * r` and maybe this is computationally better.

Reply:
note that this 'u' (I called it in the code uu because u is often used for units) is not the same as 'l' that you are using, 'l' was a distance from the black hole, while 'u' is the distance from the photon injection place. If mu_s=1 then l=u+r, but in a general mu_s case there is a more complicated formula, and you would need to include du/dl if you integrate over l 
The integration in u should start from 0 (and go to infinity) and similar like in your mu_s=1 case it is better to do it in log scale, however then you have to be careful with the starting value. And note that if the photon starts far away from the reprocessing material even a small relative step in u can considerably change the distance and the angles, hence I start from 1.e-5 to be on a safe side w.r.t. all reasonable cases of absorption.